### PR TITLE
Add /api/v1/status/websockets

### LINKED ISSUE #679: Add a status endpoint to view currently connected websocket clients

### PR DESCRIPTION
**Pull request by awrichar**
Fixes #679

### CODE REVIEW

File: internal/events/websockets/websockets.go [hunk 1]
```diff
@@ -152,3 +152,28 @@ func (ws *WebSockets) WaitClosed() {
 		ws.waitClose()
 	}
 }
+
+func (ws *WebSockets) GetStatus() *fftypes.WebSocketStatus {
+	status := &fftypes.WebSocketStatus{
+		Enabled:     true,
+		Connections: make([]*fftypes.WSConnectionStatus, 0),
+	}
+	for _, c := range ws.connections {
```
Comment (nguyer) [inline internal/events/websockets/websockets.go:161]: Do we need to lock here to avoid a read while another goroutine is writing?

**Review by nguyer — COMMENTED** [review]
Looks good. One question though on a potential concurrency issue.

**Comment by nguyer** [comment]
And there are also merge conflicts to sort before merging

File: internal/events/websockets/websockets.go (hunk 1 above)
Comment (awrichar) [reply internal/events/websockets/websockets.go]: Yea, good call. Added that.

File: internal/apiserver/route_get_status_websockets.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+var getStatusWebSockets = &oapispec.Route{
+	Name:            "getStatusWebSockets",
+	Path:            "status/websockets",
+	Method:          http.MethodGet,
+	PathParams:      nil,
+	QueryParams:     nil,
+	FilterFactory:   nil,
+	Description:     coremsgs.APIMessageTBD,
+	JSONInputValue:  nil,
+	JSONOutputValue: func() interface{} { return []*fftypes.WebSocketStatus{} },
```
Comment (peterbroadhurst) [inline internal/apiserver/route_get_status_websockets.go:36]: According to the test this is `&fftypes.WebSocketStatus{}` now

File: internal/events/websockets/websockets.go [hunk 3]
```diff
@@ -138,3 +138,35 @@ func (ws *WebSockets) WaitClosed() {
...
 }
+
+func (ws *WebSockets) GetStatus() *fftypes.WebSocketStatus {
+	status := &fftypes.WebSocketStatus{
+		Enabled:     true,
+		Connections: make([]*fftypes.WSConnectionStatus, 0),
+	}
+
+	ws.connMux.Lock()
+	connections := ws.connections
```
Comment (peterbroadhurst) [inline internal/events/websockets/websockets.go:149]: I don't think on its own this gives a safe copy.
Think this has to be:

```
wc.mux.Lock()
connections = make([]connection, len(ws.connections))
for i, c := range ws.connections {
  connections[i] = c
}
wc.mux.Unlock()
```

**Pull request merged**